python/streamtracer/streamline.py

```diff
@@ -171,35 +187,56 @@ class StreamTracer:
 
     Parameters
     ----------
-    max_steps : int
+    max_steps : `int`
         Number of steps available for each line. The maximum number of points
         on a single stream line is ``max_steps``.
-    step_size : float
+    step_size : `float`
         Step size as a the fraction of cell size.
-    cyclic : [bool, bool, bool], optional
-        Whether to have cyclic boundary conditions in each dimension.
+    """
 
-    Attributes
-    ----------
-    xs : array of (n, 3) arrays
-        An array of the streamlines, which in general can have varying
+    def __init__(self, max_steps, step_size):
+        self.max_steps = max_steps
+        self.ds = step_size
+        self.xs = None
+
+    @property
+    def xs(self):
+        """
+        An array of the streamlines each with shape ``(n,3)``, which in general can have varying
         numbers of points.
-    ROT : integer array
-        Reason(s) of termination. Shape ``len(xs)`` if traced in one direction,
+        """
+        return self._xs
+
+    @xs.setter
+    def xs(self, val):
+        self._xs = val
```

Comment:
Given the change [here](https://github.com/sunpy/streamtracer/pull/172/files#diff-7a49ca09f01796352cc0e61289b8b299fe4272203de545d5ede0aa9bb9a3e92fR308) shouldn't this be
```suggestion
        self._xs = np.array(val)
```
to maintain consistency?

Reply:
No, I don't think so. In the change below, I explicitly do not cast this as an array because each streamline can have a different number of points. That's why it's a list of arrays rather than a higher-dimensional array.